src/app/areas/areas.component.ts

```diff
@@ -24,7 +24,8 @@ export class AreasComponent implements OnInit {
 
   add() {
     const area = new Area(this.areaService);
-    this.areaService.add(area);
```

Comment:
Why you removed possibility to add Area?

Reply:
on it

Reply:
is that working for you?
I think in that case, you can't add any area. I mean when you click the + button, adding area won't work for you.